Laboratory/C++/gen/jazz.hpp

```diff
@@ -0,0 +1,156 @@
+#include <cstddef>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+#include "bebop.hpp"
+
+enum class Instrument {
+  Sax = 0,
+  Trumpet = 1,
+  Clarinet = 2,
+};
+
+struct Musician {
+  std::string name;
+  Instrument plays;
+
+  static std::unique_ptr<std::vector<uint8_t>> encode(const Musician &message) {
+    auto &writer = bebop::BebopWriter::instance();
+    Musician::encodeInto(message, writer);
+    return std::move(writer.buffer());
+  }
+
+  static void encodeInto(const Musician &message, bebop::BebopWriter &view) {
```

Comment:
How come you call the BebopWriter (and BebopReader) variable `view` instead of `writer` (or `reader`)?